seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/source/ClickhouseSource.java

```diff
@@ -17,70 +17,217 @@
+                                .collect(
+                                        Collectors.toMap(
+                                                Map.Entry::getKey,
+                                                entry -> entry.getValue().unwrapped().toString()))
+                        : null;
+
+        servers =
+                ClickhouseUtil.createNodes(
+                        config.getString(HOST.key()),
+                        config.getString(DATABASE.key()),
+                        config.getString(SERVER_TIME_ZONE.key()),
+                        config.getString(USERNAME.key()),
+                        config.getString(PASSWORD.key()),
+                        customConfig);
+
+        ClickHouseNode currentServer =
+                servers.get(ThreadLocalRandom.current().nextInt(servers.size()));
+
+        ReadonlyConfig readonlyConfig = ReadonlyConfig.fromConfig(config);
+        String sql = readonlyConfig.getOptional(SQL).orElse(null);
+
+        if (readonlyConfig.getOptional(TABLE_LIST).isPresent()) {
+            readonlyConfig.get(TABLE_LIST).stream()
+                    .map(ReadonlyConfig::fromMap)
+                    .forEach(
+                            conf -> {
+                                String confSql = conf.getOptional(SQL).get();
+                                SeaTunnelRowType clickhouseRowType =
+                                        getClickhouseRowType(currentServer, confSql);
+                                TablePath tablePath =
+                                        TablePath.of(conf.getOptional(TABLE_PATH).orElse(""));
+                                CatalogTable catalogTable =
+                                        createCatalogTable(clickhouseRowType, tablePath);
+
+                                ClickhouseCatalogConfig clickhouseCatalogConfig =
+                                        new ClickhouseCatalogConfig();
+                                clickhouseCatalogConfig.setSql(confSql);
+                                clickhouseCatalogConfig.setCatalogTable(catalogTable);
+                                tableClickhouseCatalogConfigMap.put(
+                                        tablePath, clickhouseCatalogConfig);
+                            });
+        } else {
+            SeaTunnelRowType clickhouseRowType = getClickhouseRowType(currentServer, sql);
+            CatalogTable catalogTable =
+                    CatalogTableUtil.getCatalogTable(defaultTablePath, clickhouseRowType);
+
+            ClickhouseCatalogConfig clickhouseCatalogConfig = new ClickhouseCatalogConfig();
+            clickhouseCatalogConfig.setCatalogTable(catalogTable);
+            clickhouseCatalogConfig.setSql(sql);
+            tableClickhouseCatalogConfigMap.put(
+                    TablePath.of(defaultTablePath), clickhouseCatalogConfig);
+        }
+    }
+
+    private CatalogTable createCatalogTable(SeaTunnelRowType rowType, TablePath tablePath) {
```

Comment:
why not use `CatalogTableUtil.convertDataTypeToCatalogTables()`

Reply:
![image](https://github.com/user-attachments/assets/c08b1b68-6b86-44de-875b-0288ab778690)
I think it's an outdated one. Is it okay to keep using it?